sources/Algo/src/DynModelFilterAlgorithm.cpp

```diff
@@ -0,0 +1,121 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, you can obtain one at http://mozilla.org/MPL/2.0/.
+// SPDX-License-Identifier: MPL-2.0
+//
+
+#include "DynModelFilterAlgorithm.h"
+#include "Constants.h"
+#include "Log.h"
+
+namespace dfl {
+namespace algo {
+
+void
+DynModelFilterAlgorithm::removeRpclInGeneratorsAndSvcIfMissingConnexionToSvc() {
+  std::vector<DynamicModelDefinition::DynModelId> svcToRemove;
+  for (const auto& dynamicModel : dynamicModels_) {
+    if (dynamicModel.second.lib == dfl::common::constants::svcModelName) {
+      bool isSVCconnectedToUMeasurement = checkIfSVCconnectedToUMeasurement(dynamicModel.second);
+      if (!isSVCconnectedToUMeasurement) {
+        svcToRemove.push_back(dynamicModel.second.id);
+        for (const DynamicModelDefinition::MacroConnection& generatorConnectedToSVC : dynamicModel.second.nodeConnections) {
+          const DynamicModelDefinition::MacroConnection::ElementId& genId = generatorConnectedToSVC.connectedElementId;
+          auto foundGen = std::find_if(generators_.begin(), generators_.end(),
+                                        [&genId](const algo::GeneratorDefinition& genDefinition) {
+                                          return genDefinition.id == genId;
+                                        });
+          if (foundGen != generators_.end()) {
+            foundGen->removeRpclFromModel();
+          }
+        }
+      }
+    }
+  }
+  for (const DynamicModelDefinition::DynModelId& svcId : svcToRemove) {
+    dynamicModels_.erase(svcId);
+  }
+}
+
+bool
+DynModelFilterAlgorithm::checkIfSVCconnectedToUMeasurement(const algo::DynamicModelDefinition& svcModel) const {
+  bool isSVCconnectedToUMeasurement = false;
+  const std::map<std::string, inputs::AssemblingDataBase::DynamicAutomaton>& assemblingAutomatons = assembling_.dynamicAutomatons();
+  auto assemblingAutomatonIt = assemblingAutomatons.find(svcModel.id);
+  if (assemblingAutomatonIt != assemblingAutomatons.end()) {
+    const std::vector<inputs::AssemblingDataBase::MacroConnect>& macroConnects = assemblingAutomatonIt->second.macroConnects;
+    for (const DynamicModelDefinition::MacroConnection& nodeConnection : svcModel.nodeConnections) {
+      auto macroConnectIt = std::find_if(macroConnects.begin(), macroConnects.end(),
+                                        [&nodeConnection](const inputs::AssemblingDataBase::MacroConnect& macroconnect) {
+                                          return nodeConnection.id == macroconnect.macroConnection;
+                                        });
+      if (macroConnectIt != macroConnects.end()) {
+        const std::string singleAssociationId = macroConnectIt->id;
+        const inputs::AssemblingDataBase::SingleAssociation& singleAssociation = assembling_.getSingleAssociation(singleAssociationId);
+        if (singleAssociation.bus.is_initialized()) {
```

Comment:
wouldn't it be fore efficient to find the bus connection in the assembling and then look for it in the nodeConnections?